forui/lib/src/widgets/line_calendar/line_calendar.dart

```diff
@@ -0,0 +1,346 @@
+              widget.controller.value = widget.controller.value?.add(const Duration(days: 1));
+              _controller.animateTo(
+                _controller.offset + _size,
+                duration: const Duration(milliseconds: 100),
+                curve: Curves.easeInOut,
+              );
+              return KeyEventResult.handled;
+            } else if (event is KeyDownEvent && event.logicalKey == LogicalKeyboardKey.arrowLeft) {
+              widget.controller.value = widget.controller.value?.subtract(const Duration(days: 1));
+              _controller.animateTo(
+                _controller.offset - _size,
+                duration: const Duration(milliseconds: 100),
+                curve: Curves.easeInOut,
+              );
+              return KeyEventResult.handled;
+            }
+            return KeyEventResult.ignored;
+          },
+          child: ListView.builder(
+            controller: _controller,
+            scrollDirection: Axis.horizontal,
+            padding: EdgeInsets.zero,
+            itemExtent: _size,
+            itemBuilder: (context, index) {
+              final date = widget.start.add(Duration(days: index));
+              return Container(
+                padding: EdgeInsets.symmetric(horizontal: _style.itemPadding),
+                child: FLineCalendarTile(
+                  style: _style,
+                  controller: widget.controller,
+                  date: date,
+                  isToday: widget.today.toLocalDate() == date.toLocalDate(),
+                ),
+              );
+            },
+          ),
+        ),
+      );
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+}
+
+/// [FLineCalendar]'s style.
+final class FLineCalendarStyle with Diagnosticable {
+  /// The horizontal padding around each calendar item.
+  final double itemPadding;
+
+  /// The box decoration for a selected date.
+  final BoxDecoration selectedDecoration;
+
+  /// The box decoration for an unselected date.
+  final BoxDecoration unselectedDecoration;
+
+  /// The color of the indicator for the selected current date.
+  final Color selectedCurrentDateIndicatorColor;
+
+  /// The color of the indicator for the unselected current date.
+  final Color unselectedCurrentDateIndicatorColor;
+
+  /// The content's style.
+  final FLineCalendarContentStyle content;
+
+  /// Creates a [FLineCalendarStyle].
+  const FLineCalendarStyle({
+    required this.itemPadding,
+    required this.selectedCurrentDateIndicatorColor,
+    required this.unselectedCurrentDateIndicatorColor,
+    required this.selectedDecoration,
+    required this.unselectedDecoration,
+    required this.content,
+  });
+
+  /// Creates a [FCardStyle] that inherits its properties from [colorScheme] and [typography].
+  FLineCalendarStyle.inherit({
+    required FColorScheme colorScheme,
+    required FTypography typography,
+    required FStyle style,
+  })  : itemPadding = 6.5,
```

Comment:
itemPadding should be a default parameter in FFLineCalendarStyle.new. `FLineCalendarStyle.inherit` should use a redirecting constructor (`this(....)`) instead.

We also need to assert that itemPadding is positive.